src/actions/graph.ts

```diff
@@ -106,6 +107,22 @@ export type GraphAction = IInitGraph | ISetGraphNode | ISetGraphNodes | IDeleteG
 | ISetGraphConnection  | IDeleteGraphConnection | ISetGraphConnections  | IDeleteGraphConnections;
 
 
+const getSystemNodeJackNamesFromPortInfo = (jackPortsInfo: OSCQueryRNBOJackPortInfo, patcherNodes: GraphPatcherNodeRecord[]): string[] => {
+	const pNodeIds = new Set(patcherNodes.map(pn => pn.id));
+	const portNames = [
+		...(jackPortsInfo.CONTENTS?.audio?.CONTENTS?.sources?.VALUE || []),
+		...(jackPortsInfo.CONTENTS?.midi?.CONTENTS?.sources?.VALUE || []),
+		...(jackPortsInfo.CONTENTS?.audio?.CONTENTS?.sinks?.VALUE || []),
+		...(jackPortsInfo.CONTENTS?.midi?.CONTENTS?.sinks?.VALUE || [])
+	];
+
+	return portNames.reduce((result, portName) => {
+		const [nodeName] = portName.split(":");
+		if (!pNodeIds.has(nodeName)) result.push(nodeName);
+		return result;
+	}, [] as string[]);
```

Comment:
this will identify `rnbo-control` as a "system node" since it isn't in the patcher node list..
 there are some additional applications that come with jack, for instance, here is a rendering of this branch where I run `midisine` in addition to rnbo, I moved the nodes closer together to make the image smaller... anyway, `midisine` would work better as a single node with inputs and outputs 
 
![image](https://github.com/Cycling74/rnbo-runner-panel/assets/65796/0ef79373-b500-46d1-98b6-5d9969ad8881)

so I might add something to your filter that indicates the name starts with `system` ? and.. either also render these non RNBO nodes just like RNBO graph nodes but without controls.. or, punt on that for a future ticket and simply don't render them at all?

Reply:
yeah guess we can just filter by `^system` for now? Not really a huge fan of maintaining a hardcoded list of strings with special handling for the matrix of Jack options and variants and if that becomes a clear FR I wonder if there is a better way.